### ISSUE TITLE
import lead to cockroach cluster server crash, import csv abort.

### ISSUE THREAD

**Issue by lvhui465021** [cockroachdb/cockroach#43568]
when import csv with shell, cockroach reading a import sql file creates table and import data in different database. Import 10 tables in a database, the csv file of every table has between 10 and 20GB. 
through different database names  loop to do the same.

the steps:
1. start 3 node cockroach cluster.
    the cluster cockroach start arguments. My ip is 192.168.80.129. Start 3 nodes.
    a) start --insecure --listen-addr=192.168.80.129:26257 --http-addr=192.168.80.129:8080 --store=/data/node1 --external-io-dir=/home/lvhui/drdb/csv --cache=.20 --max-sql-memory=.20 --background 
    b) start --insecure --listen-addr=192.168.80.129:26258 --http-addr=192.168.80.129:8081 --store=/data/node2 --external-io-dir=/home/lvhui/drdb/csv --cache=.20 --max-sql-memory=.20 --background --join=192.168.80.129:26257
    c) start --insecure --listen-addr=192.168.80.129:26259 --http-addr=192.168.80.129:8082 --store=/data/node3 --external-io-dir=/home/lvhui/drdb/csv --cache=.20 --max-sql-memory=.20 --background --join=192.168.80.129:26257

2. create a import sql file, Mine has 10 tables to import.
3. create 10 csv file, every csv has between 10 and 20 GB.
4. write a shell to import. mine example:
`#!/bin/bash
#create database and load data
start_seconds=$(date --date="$total_starttime" +%s)
echo total_starttime=`date +'%Y-%m-%d %H:%M:%S'` >> import.log
for i in `seq 1 50`
do
   echo "====================${i}========================" >> import.log
        echo starttime=`date +'%Y-%m-%d %H:%M:%S'` >>import.log
        starttime=`date +'%Y-%m-%d %H:%M:%S'`
   /home/lvhui/cockroach-v19.2.2/src/github.com/cockroachdb/cockroach/cockroach sql --insecure --host=192.168.80.129:26257 -e"create database tpcc_${i}">>import.log
   /home/lvhui/cockroach-v19.2.2/src/github.com/cockroachdb/cockroach/cockroach sql --insecure --host=192.168.80.129:26257 -d"tpcc_${i}"< /home/lvhui/cockroach-v19.2.2/temp.sql >>import.log
        endtime=`date +'%Y-%m-%d %H:%M:%S'`
        echo endtime=`date +'%Y-%m-%d %H:%M:%S'` >> import.log

        start_seconds=$(date --date="$starttime" +%s)
        end_seconds=$(date --date="$endtime" +%s)
        echo "tpcc_${i} loading time : "$((end_seconds-start_seconds))"s" >> import.log
done

echo total_endtime=`date +'%Y-%m-%d %H:%M:%S'`>> import.log
end_seconds=$(date --date="$total_endtime" +%s)
echo "Overall running time : "$((end_seconds-start_seconds))"s" >> import.log`

5. run the import shell in background.

The error message: context canceled or context deadline exceeded.
`W191227 02:50:45.652606 198 storage/node_liveness.go:559  [n1,liveness-hb] slow heartbeat took 4.8s
W191227 02:50:45.652625 198 storage/node_liveness.go:484  [n1,liveness-hb] failed node liveness heartbeat: operation "node liveness heartbeat" timed out after 4.5s:
  - context deadline exceeded
W191227 02:50:46.233296 140 storage/engine/rocksdb.go:2076  batch [2/26995/0] commit took 2.217815729s (>= warning threshold 500ms)
W191227 02:50:46.690868 130 storage/store_raft.go:514  [n1,s1,r24/1:/Table/{28-52}] handle raft ready: 4.7s [applied=0, batches=0, state_assertions=0]
W191227 02:50:46.715913 142 storage/store_raft.go:514  [n1,s1,r2/1:/System/NodeLiveness{-Max}] handle raft ready: 4.7s [applied=0, batches=0, state_assertions=0]
E191227 02:50:47.013187 12644 storage/replica_range_lease.go:339  [n1,s1,r24/1:/Table/{28-52}] context canceled
I191227 02:50:48.657087 12598 internal/client/txn.go:630  [n1] async rollback failed: context deadline exceeded
W191227 02:50:48.683822 153 storage/store_raft.go:514  [n1,s1,r11/1:/Table/1{5-6}] handle raft ready: 5.6s [applied=0, batches=0, state_assertions=0]

`

Expected Result:
Import succeed and not interrupt.

Actual Result:
Import interrupt and go to next loop. the database has less 10 table and data.

screenshots:
[https://github.com/lvhui465021/picture/blob/master/log_error.jpg](url)
the error log url:
[https://github.com/lvhui465021/picture/blob/master/cockroach.log](url)

Environment:
ubuntu 19.04 virtual machine
cpu: 6 core
memory: 10GB
disk: 10TB

**Comment by lvhui465021** [comment]
temp.sql is the import sql file.

**Comment by lvhui465021** [comment]
i use the cluster setting kv.bulk_io_write.max_rate set the speed blow my disk. but it not wroking.

**Comment by lvhui465021** [comment]
the import shell file, node 1 import tpcc_1 to tpcc_50 database, node 2 import tpcc_51 to tpcc_100 database.

**Comment by lvhui465021** [comment]
# the import shell.
#!/bin/bash
#create database and load data
start_seconds=$(date --date="$total_starttime" +%s)
echo total_starttime=`date +'%Y-%m-%d %H:%M:%S'` >> import.log
for i in `seq 1 50`
do
   echo "====================${i}========================" >> import.log
        echo starttime=`date +'%Y-%m-%d %H:%M:%S'` >>import.log
        starttime=`date +'%Y-%m-%d %H:%M:%S'`
   /home/lvhui/cockroach-v19.2.2/src/github.com/cockroachdb/cockroach/cockroach sql --insecure --host=192.168.80.129:26257 -e"create database tpcc_${i}">>import.log
   /home/lvhui/cockroach-v19.2.2/src/github.com/cockroachdb/cockroach/cockroach sql --insecure --host=192.168.80.129:26257 -d"tpcc_${i}"< /home/lvhui/cockroach-v19.2.2/temp.sql >>import.log
        endtime=`date +'%Y-%m-%d %H:%M:%S'`
        echo endtime=`date +'%Y-%m-%d %H:%M:%S'` >> import.log

        start_seconds=$(date --date="$starttime" +%s)
        end_seconds=$(date --date="$endtime" +%s)
        echo "tpcc_${i} loading time : "$((end_seconds-start_seconds))"s" >> import.log
done

echo total_endtime=`date +'%Y-%m-%d %H:%M:%S'`>> import.log
end_seconds=$(date --date="$total_endtime" +%s)
echo "Overall running time : "$((end_seconds-start_seconds))"s" >> import.log

**Comment by dt** [comment]
Hey SQL folks, I think someone is just piping inserts in here rather than using IMPORT so I think this might be a SQL issue?

**Comment by dt** [comment]
That said it looks from the logs like this is a disk stall and I’m pretty sure the kv team removed the disk stall crash or at least bumped it way up, so this might just be resolved by upgrading

**Comment by asubiotto** [comment]
> Hey SQL folks, I think someone is just piping inserts in here rather than using IMPORT so I think this might be a SQL issue?

I'm not sure about this. @lvhui465021 mentions that `temp.sql` is importing CSVs. It looks like the cluster is overloaded during that for some reason which is Bulk IO/KV. Not sure SQL Execution can do much here.

**Comment by dt** [comment]
Ah, thanks @asubiotto, I saw the `... cockroach sql < ` and thought it was another case of inserts-not-import.

If it was IMPORTs in those files, then that makes a bit more sense. looking at the logs I see the node died after detecting a disk stall, reporting that it was unable to write its log to disk for >30s. That _could_ happen if somehow IMPORT had managed to saturate the disks an incredible amount, but we have backpressure that should prevent that. Furthermore, immediately before that Rocks logged its periodic metrics that show we only added 0.061GB of SSTs in that interval -- 58mb in five or ten minutes doesn't sounds like saturation due to IMPORT but something more basic to do with the disk and storage.

My guess here is that we can't really say for sure without the hardware in question or at least more details, but I know storage (or maybe back then it was kv) folks have found some issues over the years with the disk stall detector, so maybe this will ring some bells as-is.
